model_zoo/lbann_cycgan.cpp

```diff
@@ -79,38 +83,71 @@ int main(int argc, char *argv[]) {
 
+
+    //Optionally pretrain autoencoder
+    //@todo: explore joint-train of autoencoder as alternative
+    if(ae_model != nullptr) {
+      if(master) std::cout << " Pre-train autoencoder " << std::endl;
+      const lbann_data::Model pb_model_4 = pbs[3]->model();
+      ae_model->train(pb_model_4.num_epochs());
+    }
+    
+    //Train cycle GAN
     int super_step = 1;
     int max_super_step = pb_model.super_steps();
     while (super_step <= max_super_step) {
       if (master)  std::cerr << "\nSTARTING train - discriminator (D1 & D2) models at step " << super_step <<"\n\n";
       model_1->train( super_step*pb_model.num_epochs(),pb_model_2.num_batches());
 
-      if(master) std::cout << " Copy all trained weights from model1 to model2 and train/freeze as appropriate " << std::endl;
+      if(master) std::cout << " Copy all trained weights from discriminator to G1 and train/freeze as appropriate " << std::endl;
       auto model1_weights = model_1->get_weights();
       model_2->copy_trained_weights_from(model1_weights);
       if (master) std::cerr << "\n STARTING train - G1 solver model at step " << super_step << " \n\n";
       model_2->train( super_step*pb_model_2.num_epochs(),pb_model_2.num_batches());
       // Evaluate model on test set
       model_2->evaluate(execution_mode::testing);
 
-      if(master) std::cout << " Copy all trained weights from model1 to model3 and train/freeze as appropriate " << std::endl;
+      if(master) std::cout << " Copy all trained weights from discriminator to G2 and train/freeze as appropriate " << std::endl;
       model_3->copy_trained_weights_from(model1_weights);
       if (master) std::cerr << "\n STARTING train - G2 solver model at step " << super_step << " \n\n";
       model_3->train( super_step*pb_model_3.num_epochs(),pb_model_3.num_batches());
       // Evaluate model on test set
       model_3->evaluate(execution_mode::testing);
 
-      if(master) std::cout << " Update generator1 weights " << std::endl;
+      if(master) std::cout << " Update G1 weights " << std::endl;
       auto model2_weights = model_2->get_weights();
       model_1->copy_trained_weights_from(model2_weights);
-      if(master) std::cout << " Update generator2 weights " << std::endl;
+      if(master) std::cout << " Update G2 weights " << std::endl;
       auto model3_weights = model_3->get_weights();
       model_1->copy_trained_weights_from(model3_weights);
+      
+      //Optionally evaluate on pretrained autoencoder
+      if(ae_model != nullptr && ae_proxy_model != nullptr){
+        if(master) std::cout << " Copy trained weights from autoencoder to autoencoder proxy" << std::endl;
+        auto ae_weights = ae_model->get_weights();
+        ae_proxy_model->copy_trained_weights_from(ae_weights);
+        if(master) std::cout << " Copy needed activations from cycle GAN" << std::endl;
+        auto model2_layers = model_2->get_layers();
+        ae_proxy_model->copy_activations_from(model2_layers);
```

Comment:
I'm unsure if this will do what we're expecting. Every layer overwrites its activations during its forward prop step, so I suspect that the copies done here will be overwritten when `evaluate` is called.

Reply:
This isn't entirely relevant here since we only evaluate on `ae_proxy_model`, but the `copy_activations_from` function can completely disrupt our automatic differentiation.

Reply:
@timmoon10 Changes have been made based on your feedback.